lib/compress/zstd_compress_sequences.h

```diff
@@ -0,0 +1,420 @@
+/*
+ * Copyright (c) 2016-present, Yann Collet, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under both the BSD-style license (found in the
+ * LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ * in the COPYING file in the root directory of this source tree).
+ * You may select, at your option, one of the above-listed licenses.
+ */
+
+#ifndef ZSTD_COMPRESS_SEQUENCES_H
+#define ZSTD_COMPRESS_SEQUENCES_H
+
+/**
+ * -log2(x / 256) lookup table for x in [0, 256).
+ * If x == 0: Return 0
+ * Else: Return floor(-log2(x / 256) * 256)
+ */
+static unsigned const kInverseProbabilityLog256[256] = {
```

Comment:
minor note on code size :
this will create an instance of the table in each file where it's included and used.

Reply:
I see. I can create .c file for zstd_compress_sequences and move this variable there. I have a few questions though.

1. Do you think my approach of organizing code for zstd_compress_sequences and zstd_compress_literals make sense?

2. What are the rules of thumb for writing the function implementation within header file vs. just writing function signature in header file and implementing it in c file? In zstd_compress_interal.h I see some functions are implemented directed in the header file (like `ZSTD_fWeight()`) while other functions like `ZSTD_resetSeqStore()` is only declared in header file and implemented in c file.